boards/arm/96b_carbon/Kconfig.board

```diff
@@ -7,5 +7,6 @@
 
 config BOARD_96B_CARBON
 	bool "96Boards Carbon (STM32F401)"
+	select HAS_DTS_SPI
```

Comment:
Why should we enable HAS_DTS_SPI per board? When SPI_STM32 is enabled, it could select HAS_DTS_SPI in Kconfig.

Reply:
So it's fine to select HAS_DTS_SPI from the config HAS_DTS_SPI_DEVICE? First time I mess with dts.

Reply:
I was not clear. I mean to select HAS_DTS_SPI in `menuconfig SPI_STM32`. We use HAS_DTS_SPI_DEVICE for the devices connected on the SPI bus (e.g. sensors, bluetooth modules).

Reply:
yes, realized that while fixing the rebase issue